File: src/androidTest/java/pt/joaomneto/titancompanion/TestTFOD.java
package pt.joaomneto.titancompanion;


import android.support.test.filters.LargeTest;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import pt.joaomneto.titancompanion.consts.FightingFantasyGamebook;
import pt.joaomneto.titancompanion.util.LocaleHelper;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withSpinnerText;
import static org.hamcrest.Matchers.containsString;
import static pt.joaomneto.titancompanion.consts.FightingFantasyGamebook.THE_FOREST_OF_DOOM;

@LargeTest
@RunWith(AndroidJUnit4.class)
public class TestTFOD extends TestTWOFM{

    @Override
    protected FightingFantasyGamebook getGamebook(){
        return THE_FOREST_OF_DOOM;
    }

    @Override
    protected void assertCorrectPotionDosage() {
        String language = LocaleHelper.getLanguage(mActivityTestRule.getActivity());
        String string = getString("fr".equals(language)?R.string.potionTwoDoses:R.string.potionOneDose);
        onView(withId(R.id.potionDosesSpinner)).check(matches(withSpinnerText(containsString(string))));
    }

}

File: src/androidTest/java/pt/joaomneto/titancompanion/phase1/TestTFOD.kt
package pt.joaomneto.titancompanion.phase1

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withSpinnerText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import org.hamcrest.Matchers.containsString
import org.junit.runner.RunWith
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.consts.FightingFantasyGamebook.THE_FOREST_OF_DOOM
import pt.joaomneto.titancompanion.util.LocaleHelper

@LargeTest
@RunWith(AndroidJUnit4::class)
open class TestTFOD : TestTWOFM() {

    override val gamebook = THE_FOREST_OF_DOOM

    override fun assertCorrectPotionDosage() {
        val language = LocaleHelper.getLanguage(mActivityTestRule.activity)
        val string = getString(if ("fr" == language) R.string.potionTwoDoses else R.string.potionOneDose)
        onView(withId(R.id.potionDosesSpinner)).check(matches(withSpinnerText(containsString(string))))
    }
}
